src/main/java/cloud/fogbow/as/core/models/OneToOneMappableSystemUser.java

```diff
@@ -1,16 +1,7 @@
 package cloud.fogbow.as.core.models;
 
-import cloud.fogbow.common.models.SystemUser;
+import cloud.fogbow.common.models.CloudUser;
 
-public class OneToOneMappableSystemUser extends SystemUser {
-    private String token;
-
-    public OneToOneMappableSystemUser(String userId, String userName, String identityProviderId, String token) {
-        super(userId, userName, identityProviderId);
-        this.token = token;
-    }
-
-    public String getToken() {
-        return token;
-    }
+public interface OneToOneMappableSystemUser {
+    public CloudUser generateCloudUser();
```

Comment:
Shouldn't this interface be defined using generics? 
public interface OneToOneMappableSystemUser<T extends CloudUser> {
    public T generateCloudUser();